libcalico-go/lib/dispatcher/blocking.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dispatcher
+
+import (
+	"context"
+	"errors"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+)
+
+// BlockingDispatcher blocks on an input channel until stop,
+// and broadcasts each input message to all output channels.
+// Blocks until all output channels have consumed a message,
+// before pulling in the next input message.
+type BlockingDispatcher[T any] struct {
```

Comment:
Couple of thoughts: 
- I don't see any problem with using reflect.Select to implement this kind of logic.  "One clever trick" behind a nice, minimal API is often a good way to go.
- Note that `reflect.Select` has an inefficient slice copy inside it to protect the runtime (nbd here but can be an issue in a tight loop).
- Overall, you might be better off giving all the output channels a buffer (say length 10) and then you could do:

```
msg <- input
for _, outC := range outputs {
  switch {
  case <-ctx.Done():
    return
  case outC <- msg:
  }
}
```

That has the advantage(?) that one consumer can be 10 ahead of another before they start to get paced by the slowest consumer.